### PR TITLE
Change `torch_dtype` to str when `saved_model=True` in `save_pretrained` for TF models

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

One issue in #22731 is: the config contains `torch_type` with a torch dtype class as value. Usually, our `save_pretrained` will take care of it. But when `saved_model=True` in `save_pretrained` for TF models, it is not handled, and TF/Keras complains about it.

See the first part in [this comment](https://github.com/huggingface/transformers/issues/22731#issuecomment-1506538658)

### CODE REVIEW

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -2313,6 +2313,13 @@ def save_pretrained(
             files_timestamps = self._get_files_timestamps(save_directory)
 
         if saved_model:
+            # If `torch_dtype` is in the config with a torch dtype class as the value, we need to change it to string.
```
Comment (amyeroberts) [inline src/transformers/modeling_tf_utils.py:2316]: Nice comment :)

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding this fix!

File: src/transformers/modeling_tf_utils.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/modeling_tf_utils.py:2316]: A bit long though 😅 I think the first sentence is more than enough ;-)

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

File: src/transformers/modeling_tf_utils.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/modeling_tf_utils.py:2316]: I will change it back to the first line lone :-).

**Review by Rocketknight1 — APPROVED** [review]
Seems like a good change, thanks!

**Pull request merged**